packages/opentelemetry-core/src/trace/Span.ts

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as types from '@opentelemetry/types';
+import { randomSpanId, randomTraceId } from '../platform';
+import { isValid, INVALID_SPAN_CONTEXT } from './spancontext-utils';
+import { performance } from 'perf_hooks';
+import { TraceOptions } from '@opentelemetry/types';
+
+/**
+ * This class represents a span.
+ */
+export class Span implements types.Span {
+  private _spanContext: types.SpanContext = INVALID_SPAN_CONTEXT;
```

Comment:
Won't that always reuse the same span context for every span?

Reply:
I am trying to set some (invalid in this case) default value for `SpanContext`, constructor will set actual values for `traceId` and `spanId`.

Reply:
That's what I mean, then it means that the constructor is overriding properties on an instance that is shared at the class level. I may also be misunderstanding what this code does, is this assignment executed on every instance?

Reply:
It is 👍